docs/dev/kube-apiserver-health-check.md

```diff
@@ -0,0 +1,82 @@
+## Graceful Termination
+`kube-apiserver` in OpenShift is fronted by an external and an internal Load Balancer. This document serves as a
+guideline on how to properly configure the health check probe of the load balancers so that when a `kube-apiserver`
+instance restarts we can ensure:
+- The load balancers detect it and takes it out of service in time. No new request should be forwarded to the 
+  `kube-apiserver` instance when it has stopped listening.
+- Existing connections are not cut off hard, they are allowed to complete gracefully.
+
+## Load Balancer Health Check Probe
+`kube-apiserver` provides graceful termination support via the `/readyz` health check endpoint. When `/readyz` reports
+`ok` it indicates that the apiserver is ready to serve request(s).
```

Comment:
nit: `ok` -> `200 OK`?  I expect LBs to care about HTTP status codes and not about the response body.  And your `ok` is likely shorthand for the 200 status, but I think explicitly saying "200" (and possibly even "HTTP status `200 OK`) would make it harder to misunderstand.

Reply:
> > Thanks for the detailed doc @tkashem !
> > I think the next step will be to make this doc discoverable by linking this doc from the code that defines these healthchecks in `data/data/{aws,gcp}` ..

@abhinavdahiya I linked the doc.